sdk/client/query_test.go

```diff
@@ -158,3 +161,130 @@ func TestQueryIsBlockBabylonFinalized(t *testing.T) {
+		{"consecutive blocks with finalized partially and error", nil, &blockB.BlockHeight, []*cwclient.L2Block{&blockA, &blockB, &blockC, &blockD}},
+		{"all consecutive blocks with finalized", nil, &blockB.BlockHeight, []*cwclient.L2Block{&blockA, &blockB}},
+		{"consecutive blocks without finalized", nil, nil, []*cwclient.L2Block{&blockE, &blockF}},
+		{"consecutive blocks without finalized and with error", nil, nil, []*cwclient.L2Block{&blockE, &blockF, &blockG}},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctl := gomock.NewController(t)
+			defer ctl.Finish()
+
+			mockCwClient := mocks.NewMockICosmWasmClient(ctl)
+			mockBTCClient := mocks.NewMockIBitcoinClient(ctl)
+			mockBBNClient := mocks.NewMockIBabylonClient(ctl)
+			mockSdkClient := &SdkClient{
+				cwClient:  mockCwClient,
+				bbnClient: mockBBNClient,
+				btcClient: mockBTCClient,
+			}
+
+			mockCwClient.EXPECT().QueryIsEnabled().Return(true, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryConsumerId().Return("consumer-chain-id", nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockAWithHashTrimmed).Return([]string{"pk1", "pk2", "pk3"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockBWithHashTrimmed).Return([]string{"pk1", "pk2", "pk3"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockCWithHashTrimmed).Return([]string{"pk1"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockDWithHashTrimmed).Return([]string{"pk3"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockEWithHashTrimmed).Return([]string{"pk1"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockFWithHashTrimmed).Return([]string{"pk2"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockGWithHashTrimmed).Return([]string{"pk3"}, nil).AnyTimes()
+			mockCwClient.EXPECT().QueryListOfVotedFinalityProviders(&blockHWithHashTrimmed).Return([]string{"pk1", "pk2", "pk3"}, nil).AnyTimes()
+
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockA.BlockTimestamp).Return(uint64(111), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockB.BlockTimestamp).Return(uint64(111), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockC.BlockTimestamp).Return(uint64(111), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockD.BlockTimestamp).Return(uint64(111), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockE.BlockTimestamp).Return(uint64(112), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockF.BlockTimestamp).Return(uint64(112), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockG.BlockTimestamp).Return(uint64(112), nil).AnyTimes()
+			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(blockH.BlockTimestamp).Return(uint64(113), nil).AnyTimes()
```

Comment:
```suggestion
			mockBTCClient.EXPECT().GetBlockHeightByTimestamp(gomock.Any()).Return(uint64(113), nil).AnyTimes()
```

This can replace L271-278

Reply:
we need every l2 block to return its btc height

Reply:
Ah I see, some require height 112 and some require height 113. All good them :see_no_evil: